### PR TITLE
MODFEE-188 Create endpoint to fetch sources associated with service point

### PR DESCRIPTION
**Pull request by alexanderkurash**
Resolves [MODFEE-188](https://issues.folio.org/browse/MODFEE-188)

### Purpose
UI needs a list of sources (users) for fee/fine actions created at a certain service point to be able to create a checkbox with sources (one of the parameters for Cash drawer reconciliation report).

### Approach
Add an endpoint that would return a list of sources.

### CODE REVIEW

File: ramls/feefine-reports.raml [hunk 1]
```diff
@@ -72,3 +74,27 @@ traits:
...
+          200:
+            body:
+              application/json:
+                schema: cashDrawerReconciliationReportSources
+                example: !include examples/cashDrawerReconciliationReportSources.sample
+          422:
+            description: "Unprocessable entity"
+            body:
+              text/plain:
+                example: "Invalid createdAt parameter"
```
Comment (alexanderkurash) [inline ramls/feefine-reports.raml:95]: Remove example

File: ramls/reports/cashDrawerReconciliationReportSourcesRequest.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Cash drawer reconciliation report sources request",
+  "description": "Represents a request for fetching sources (users that performed activities) required for cash drawer reconciliation report",
+  "type": "object",
+  "properties": {
+    "createdAt": {
+      "type": "string",
+      "description": "A globally unique (UUID) identifier for the service point",
```
Comment (alexanderkurash) [inline ramls/reports/cashDrawerReconciliationReportSourcesRequest.json:9]: Change description

File: src/test/java/org/folio/rest/impl/CashDrawerReconciliationReportTest.java [hunk 3]
```diff
@@ -218,12 +254,11 @@ private void requestAndCheck(CashDrawerReconciliationReport report,
...
     int numberOfEntries = entries.size();
 
-    ValidatableResponse response = requestReport(startDate, endDate, createdAt, sources)
+    requestReport(startDate, endDate, createdAt, sources)
       .then()
       .statusCode(HttpStatus.SC_OK)
-      .body("reportData", iterableWithSize(numberOfEntries));
-
-    cashDrawerReconciliationReportMatcher(report);
+      .body("reportData", iterableWithSize(numberOfEntries))
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/impl/CashDrawerReconciliationReportTest.java:260]: Do we need this check?

File: src/test/java/org/folio/test/support/matcher/ReportMatcher.java [hunk 4]
```diff
@@ -109,4 +114,11 @@ private ReportMatcher() {}
       hasJsonPath("totalCount", is(entry.getTotalCount()))
     );
   }
+
+  public static Matcher<? super Object> cashDrawerReconciliationReportSourcesMatcher(
+    CashDrawerReconciliationReportSources cashDrawerReconciliationReportSources) {
+
+//    return hasJsonPath("sources", contains(cashDrawerReconciliationReportSources.getSources()));
```
Comment (alexanderkurash) [inline src/test/java/org/folio/test/support/matcher/ReportMatcher.java:121]: Remove comment

File: src/test/java/org/folio/test/support/matcher/ReportMatcher.java [hunk 5]
```diff
@@ -51,14 +53,17 @@ private ReportMatcher() {}
...
-        cashDrawerReconciliationReport.getReportData().stream()
-          .map(ReportMatcher::cashDrawerReconciliationReportEntryMatcher)
-          .collect(Collectors.toList()))),
+      hasJsonPath("reportData",
+        cashDrawerReconciliationReport.getReportData().isEmpty() ?
+          is(List.of()) :
+          contains(
+            cashDrawerReconciliationReport.getReportData().stream()
+              .map(ReportMatcher::cashDrawerReconciliationReportEntryMatcher)
+              .collect(Collectors.toList()))),
```
Comment (alexanderkurash) [inline src/test/java/org/folio/test/support/matcher/ReportMatcher.java:66]: Reformat

File: src/test/java/org/folio/rest/impl/CashDrawerReconciliationReportTest.java [hunk 6]
```diff
@@ -208,6 +215,35 @@ public void validReportWhenPaymentsExist() {
           buildReportTotalsEntry(FEE_FINE_OWNER_TOTALS, "16.00", "4")))));
   }
 
+  @Test
+  public void sources() {
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/impl/CashDrawerReconciliationReportTest.java:219]: Rename

File: src/test/java/org/folio/rest/impl/CashDrawerReconciliationReportTest.java [hunk 7]
```diff
@@ -208,6 +215,35 @@ public void validReportWhenPaymentsExist() {
...
+    Account account1 = charge(USER_ID_1, 10.0, FEE_FINE_TYPE_1, null, OWNER_ID_1, OWNER_1);
+    Account account2 = charge(USER_ID_1, 10.0, FEE_FINE_TYPE_2, null, OWNER_ID_1, OWNER_1);
+    Account account3 = charge(USER_ID_2, 10.0, FEE_FINE_TYPE_2, null, OWNER_ID_2, OWNER_2);
+
+    createAction(USER_ID_1, 1, account1, withTenantTz("2020-01-01 12:00:00"),
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/impl/CashDrawerReconciliationReportTest.java:224]: Do we need withTenantTz?

File: src/test/java/org/folio/rest/impl/CashDrawerReconciliationReportTest.java (hunk 6 above)
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/impl/CashDrawerReconciliationReportTest.java:219]: Add more tests - 1) sources for a fee/fine action with a different service point ID

File: src/test/java/org/folio/rest/utils/ReportResourceClient.java [hunk 8]
```diff
@@ -45,6 +45,17 @@ public Response getCashDrawerReconciliationReport(String startDate, String endDa
       createdAt, sources), expectedStatus);
   }
 
+  public Response getCashDrawerReconciliationReportSources(String createdAt) {
+
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/utils/ReportResourceClient.java:49]: Blank line

File: src/main/java/org/folio/rest/repository/FeeFineActionRepository.java [hunk 9]
```diff
@@ -197,6 +197,33 @@ public FeeFineActionRepository(Map<String, String> headers, Context context) {
     return promise.future().map(this::mapToFeeFineActionsAndAccounts);
   }
 
+  public Future<List<String>> findPaymentSources(String createdAt, int limit) {
+    Tuple params = Tuple.of(limit);
+    List<String> conditions = new ArrayList<>();
+
+    params.addString(PAY.getFullResult());
+    params.addString(PAY.getPartialResult());
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/repository/FeeFineActionRepository.java:205]: Extract parameter and rename method

**Review by roman-barannyk — APPROVED** [review]

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**